src/repository/index.ts

```diff
@@ -197,6 +217,12 @@ export default class Repository extends EventEmitter implements EventEmitter {
   }
 
   async start(): Promise<void> {
+    // Create EventSource if streaming mode and none provided
+    if (this.mode.type === 'streaming' && !this.eventSource) {
```

Comment:
this is a main reason I want to have another PR with separate start strategy for polling and streaming (to remove those if statements)